.github/workflows/release-please.yml

```diff
@@ -0,0 +1,46 @@
+on:
+  push:
+    branches:
+      - dev
+
+permissions:
+  contents: write
+  pull-requests: write
+
+name: release-please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v4
+        with:
+          # this assumes that you have created a personal access token (PAT)
+          token: ${{ secrets.GITHUB_RELEASE_PLEASE_TOKEN }}
+          # optional. customize path to release-please-config.json
+          config-file: release-please-config.json
+          # optional. customize path to .release-please-manifest.json
+          manifest-file: .release-please-manifest.json
+
+      # The logic below handles the npm publication:
+      - uses: actions/checkout@v4
+        # these if statements ensure that a publication only occurs when
+        # a new release is created:
+        if: ${{ steps.release.outputs.release_created }}
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: '^8.15'
+        if: ${{ steps.release.outputs.release_created }}
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          registry-url: 'https://registry.npmjs.org/'
+          cache: 'pnpm'
+        if: ${{ steps.release.outputs.release_created }}
```

Comment:
`if` statements in [github actions don't need wrapping in `${{` syntax btw](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif:~:text=matrix%20is%20applied.-,When%20you%20use%20expressions%20in%20an%20if%20conditional%2C%20you%20can%2C%20optionally%2C%20omit%20the%20%24%7B%7B%20%7D%7D,-expression%20syntax%20because)